OsmAnd/src/net/osmand/plus/routing/data/AnnounceTimeDistances.java

```diff
@@ -174,28 +174,35 @@ public boolean isTurnStateNotPassed(float currentSpeed, double dist, int turnTyp
 	}
 
 	private boolean isDistanceLess(float currentSpeed, double dist, double leadDist) {
-		return isDistanceLess(currentSpeed, dist, leadDist, (float) leadDist / DEFAULT_SPEED);
+		return isDistanceLess(currentSpeed, dist, leadDist, 0.0f);
```

Comment:
I need a way to identify the "TURN_NOW" case. And the easiest is to check if we called _isDistanceLess(...)_ with a leadTime, as for all but the TURN_NOW case we don't.

So using interim 0 here instead of a separate switch.

Reply:
That means you need to create separate method for Turn now and not reuse this, this is a careful refactoring needed. But I still didn't understand the final concept...
Minimum distance is needed anyway, so why then it needs to be lower